### PR TITLE
#205: Release process is broken

### LINKED ISSUE #205: Release process is broken
## Describe the bug
The release action did not complete to get out Atum 0.2

## To Reproduce
Hard to reproduce.
See log: https://github.com/AbsaOSS/atum-service/actions/runs/9363398772/job/25774102738#step:4:194

## Expected behavior
Release succeeds.

### PR DESCRIPTION
**Pull request by benedeki**
* removed incompatible settings from sbt project setup
* separated Sonatype and GitHub release, as they use different Java versions
* updated build.yml for newest action versions

Closes #205

### CODE REVIEW

File: project/Setup.scala [hunk 1]
```diff
@@ -43,14 +43,14 @@ object Setup {
...
-  val serverAndDbScalacOptions: Seq[String] = Seq("-release", "11", "-Ymacro-annotations")
+  val serverAndDbScalacOptions: Seq[String] = Seq("-Ymacro-annotations")
 
   val clientJavacOptions: Seq[String] = Seq("-source", "1.8", "-target", "1.8", "-Xlint")
   def clientScalacOptions(scalaVersion: Version): Seq[String] = {
     if (scalaVersion >= scala213) {
       Seq("-release", "8", "-Ymacro-annotations")
     } else {
-      Seq("-target", "8", "-release", "8")
+      Seq("-release", "8", "-target:8")
```
Comment (lsulak) [inline project/Setup.scala:53]: is it really needed together? `"-target:8"`

**Review by lsulak — COMMENTED** [review]
LGTM although I asked 2 questions - otherwise I built it locally and it works

File: project/Setup.scala [hunk 2]
```diff
@@ -43,14 +43,14 @@ object Setup {
   val commonScalacOptions: Seq[String] = Seq("-unchecked", "-deprecation", "-feature", "-Xfatal-warnings")
 
   val serverAndDbJavacOptions: Seq[String] = Seq("-source", "11", "-target", "11", "-Xlint")
-  val serverAndDbScalacOptions: Seq[String] = Seq("-release", "11", "-Ymacro-annotations")
+  val serverAndDbScalacOptions: Seq[String] = Seq("-Ymacro-annotations")
```
[the remark this answers is not in the document]
Comment (lsulak) [reply project/Setup.scala:46]: oh okay, I can see this in err logs: `'11' is not a valid choice for '-release'`

**Review by lsulak — DISMISSED** [review]

File: .github/workflows/release_publish.yml [hunk 3]
```diff
@@ -20,21 +20,47 @@ on:
...
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: coursier/cache-action@v5
+
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v14
+        with:
+          java-version: "adopt@1.11.0-11"
+      - name: Run sbt ci-release (produces jar as well)
```
Comment (lsulak) [inline .github/workflows/release_publish.yml:61]:
```suggestion
      - name: Run assembly (produces jar as well)
```

File: project/Setup.scala (hunk 1 above)
Comment (benedeki) [reply project/Setup.scala:53]: 🤷

**Review by miroslavpojer — APPROVED** [review]

**Pull request merged**